docs/modules/string.md

```diff
@@ -0,0 +1,329 @@
+
+Manage strings (text), allowing easy transformation and searching within the text.
+
+```{admonition} Technical Limitation
+:class: warning
+
+Due to the way string are handled, we don't support using `"` you need to use the `\"` instead.
+
+```
+
+---
+
+## 🔧 Functions
+
+You can find below all functions available in this module.
+
+---
+
+### Concat
+
+```{function} #bs.string:concat
+
+Merge multiple strings from the input list into a single string.
+
+:Inputs:
+  **Storage `bs:in string.concat.list`**: {nbt}`string` List of strings to merge
+
+:Outputs:
+  **Storage `bs:out string.concat`**: {nbt}`string` The merged string
+```
+
+_Merge `Hello ` and `World`:_
+
+```mcfunction
+data modify storage bs:in string.concat.list set value ["Hello ","World"]
+function #bs.string:concat
+tellraw @a [{"text":"The merged string is \""},{"storage":"bs:out","nbt":"string.concat"},{"text":"\""}]
+```
+
+> **Credits**: Aure31
+
+---
+
+### Convert Case
+
+:::::{tab-set}
+::::{tab-item} Uppercase
+
+```{function} #bs.string:upper
+
+Convert text to uppercase.
+
+:Inputs:
+  **Storage `bs:in string.upper.str`**: {nbt}`string` The string to convert to uppercase.
+
+:Outputs:
+  **Storage `bs:out string.upper`**: {nbt}`string` The uppercase string.
+```
+
+_Convert `hello world` to uppercase:_
+
+```mcfunction
+data modify storage bs:in string.upper.str set value "hello world"
+function #bs.string:upper
+tellraw @a [{"text":"The uppercase string is \""},{"storage":"bs:out","nbt":"string.upper"},{"text":"\""}]
+```
+
+::::
+::::{tab-item} Lowercase
+
+```{function} #bs.string:lower
+
+Convert text to lowercase.
+
+:Inputs:
+  **Storage `bs:in string.lower.str`**: {nbt}`string` The string to convert to lowercase.
+
+:Outputs:
+  **Storage `bs:out string.lower`**: {nbt}`string` The lowercase string.
+```
+
+_Convert `HELLO WORLD` to lowercase:_
+
+```mcfunction
+data modify storage bs:in string.lower.str set value "HELLO WORLD"
+function #bs.string:lower
+tellraw @a [{"text":"The lowercase string is \""},{"storage":"bs:out","nbt":"string.lower"},{"text":"\""}]
+```
+
+::::
+:::::
+
+> **Credits**: Aure31
+
+---
+
+### Find
+
+```{function} #bs.string:find
+
+Search for a substring within a string. Locate all occurrences or a specific number of matches.
+
+:Inputs:
+  **Storage `bs:in string.find`**:
+  :::{treeview}
+  - {nbt}`compound` Find data
+    - {nbt}`string` **str**: String to search in.
+    - {nbt}`string` **substr**: String to find.
+    - {nbt}`int` **occurrence**: Number of occurrences to find (0 for all).
+  :::
+
+:Outputs:
+  **Storage `bs:out string.find`**: {nbt}`list` List of indices where the substring was found.
+
+  **Return**: Number of occurrences found.
+```
+
+_Find `world` in `hello world`:_
+
+```mcfunction
+data modify storage bs:in string.find set value {str:"hello world",substr:"world",occurrence:0}
+function #bs.string:find
+tellraw @a [{"text":"Found at indices: \""},{"storage":"bs:out","nbt":"string.find"},{"text":"\""}]
```

Comment:
Put in comment the two expected results. We should be able to read the results without having to execute the command.